From bdd80539111fa8d2a0f0db98f2cf8d12f67be2f0 Mon Sep 17 00:00:00 2001
From: Scott K Logan <logans@cottsay.net>
Date: Mon, 1 Feb 2021 15:16:01 -0800
Subject: [PATCH] Suppress fortification in RPM builds

---
 rpm/template.spec.em | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/rpm/template.spec.em b/rpm/template.spec.em
index 836b2af..39cf492 100644
--- a/rpm/template.spec.em
+++ b/rpm/template.spec.em
@@ -31,6 +31,9 @@ Source0:        %{name}-%{version}.tar.gz
 %autosetup -p1
 
 %build
+# Suppress fortification, which conflicts with Mimick's -O0 requirement
+export CFLAGS="${CFLAGS:-%{optflags}} -Wp,-U_FORTIFY_SOURCE"
+
 # In case we're installing to a non-standard location, look for a setup.sh
 # in the install tree and source it.  It will set things like
 # CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
-- 
2.35.1

